### ISSUE TITLE
Overview of steps

### ISSUE THREAD

**Issue by dschwilk** [EmmaLiTsai/recoverKBTDR#3]
I'm starting an issue here to discuss the plan for overall order of steps in the dive data digitization. 

My current guess:

1. Convert x units to time using the dots. We should be able to do this without anything else being done, right? So this can happen first. Getting the arc transformation correct requires this is done first, I think.
2. Initial transformation of time axes to avoid overlap in y and allow spline smoothing. Currently this is simple transformation: `new_x = x + y*0.5`.  We need to record and hold onto the original y so this can be back transformed.  We can leave this as is for now if it works. It does have the effect of stretch the x axis which influences spline fitting.  Better for smoothing would be to do a linear transformation of the 2d data to new axes.  We can add that in if necessary -- that is a slot-in step.
3. Spline smoothing
4. Dive decomposition based on first derivative of above. Calculation of descent and ascent start and stop points needed for step 5. 
5. Transform original x (time corrected), y data using arc equation, max depth and ascent flags. 
6. Re-smooth (splines) now that we have data in correct coordinates.

**Comment by dschwilk** [comment]
On some thought, I realize that I am confused as to why we need to do the initial transformation, smoothing, and descent/ascent detection. My understanding of the device might be wrong, but why can't we just correct the x value from the raw scan data? We know arm length, arm pivot height above the y axis, and I assume that the "correct" time is where the arm intersects the y axis, so it is a few complicated triangles, but no need to know speed, max depth, etc. I must be missing something.

Here is my diagram and math. I assume that the pivot point is above the highest y value. Is this true? If not, then we need to break the problem in two depending on y value. L is arm length, and H is arm pivot point height above axis. Sorry for the extra Greek letters. When I started I thought it would require more trig but it all works with Pythogorean theorem.
![arc-arm](https://user-images.githubusercontent.com/2388939/102382459-1845cc00-3f90-11eb-9b96-11bac37f3cad.png)

I could not figure out L and H from your code and I checked the MS proposal but found no diagram of the device. You code mentions that the dimensions of the arm are "325.08 and 18.03" -- those are in units of the pixels in your scan? Hm, those do not seem to work. For my calculations to be correct, L should be around 27-28 and H around 14-15. Weird that those two numbers you have are so related 18.03^2 = 325
